src/main/scala/mesosphere/marathon/Main.scala

```diff
@@ -81,14 +87,39 @@ class MarathonApp extends App {
     with HttpEventConfiguration
     with DebugConf
 
-  override lazy val conf = new AllConf
+  lazy val conf = new AllConf
+  var serviceManager: Option[ServiceManager] = None
 
   def runDefault(): Unit = {
     log.info(s"Starting Marathon ${BuildInfo.version}")
-    run(
-      classOf[HttpService],
-      classOf[MarathonSchedulerService]
+
+    conf.afterInit()
+
+    sys.addShutdownHook(shutdownAndWait())
+
+    lazy val injector = Guice.createInjector(modules().asJava)
+    val services = List(
+      injector.getInstance(classOf[HttpService]),
+      injector.getInstance(classOf[MarathonSchedulerService])
     )
+    val serviceManager = new ServiceManager(services.asJava)
+    this.serviceManager = Some(serviceManager)
+
+    serviceManager.startAsync
+
+    Try { serviceManager.awaitHealthy() } match {
+      case Success(_) =>
+        log.info("All services up and running.")
+      case Failure(f) =>
+        log.fatal(s"Failed to start all services. Services by state: ${serviceManager.servicesByState()}", f)
+        shutdownAndWait()
+        System.exit(1)
+    }
+  }
+
+  private def shutdownAndWait(): Unit = {
+    serviceManager.foreach(_.stopAsync())
+    serviceManager.foreach(_.awaitStopped())
```

Comment:
you should set `serviceManager` to `None` afterwards to prevent shutting it down twice (via the shutdown hook and directly), even though it does not appear to cause harm.

Some kind of logging "Now shutting down." "Shutdown completed." or something like that would be nice.